src/reducers/jobListingReducer.ts

```diff
@@ -0,0 +1,53 @@
+import type { JobData, JobTransformed } from '../types/JobListing';
+import {SearchFilterAction} from '../types/JobListing'
+
+export const jobListingReducer = (
+  state: JobData,
+  action: SearchFilterAction
+): JobData => {
+  console.log(state, action);
+  let searchFilters = state.searchFilters;
+  let filteredJobListings = state.filteredJobListings;
+  console.log({ state });
+  switch (action.type) {
+    case 'CLEAR_FILTER':
+      return {
+        ...state,
+        filteredJobListings: [],
+        searchFilters: []
+      };
+    case 'ADD_FILTER':
+      if (!searchFilters.includes(action.value)) {
+        searchFilters = [...searchFilters, action.value];
+        const jobListingData = state.jobData.filter((job: JobTransformed) =>
+          job.tags.includes(action.value)
+        );
+        const allJobListings = new Set([
+          ...jobListingData,
+          ...filteredJobListings
+        ]);
+        filteredJobListings = Array.from(allJobListings);
+      }
+
+      console.log({ filteredJobListings });
+      return {
+        ...state,
+        searchFilters,
+        filteredJobListings
+      };
```

Comment:
This could be simplified to the following:

```tsx
// precondition, early return
if (state.searchFilters.includes(action.value)) return state;

return {
  ...state,
  searchFilters: [...state.searchFilters, action.value],
  filteredJobListings: Array.from(
    new Set([
      ...state.jobData.filter((job: JobTransformed) =>
        job.tags.includes(action.value)
      ),
      ...state.filteredJobListings,
    ])
  ),
};
```

`let` variables can be reassigned, so I always try to use `const` as much as possible. In this case, even `const` is unnecessary since it all can be done inlined in less code.